R/open_curly_linter.R

```diff
@@ -9,6 +9,7 @@
 #'   <https://style.tidyverse.org/syntax.html#indenting>
 #' @export
 open_curly_linter <- function(allow_single_line = FALSE) {
```

Comment:
Could you explain why not call `brace_linter()` from here? I have an idea but it would be good to have it written out.

I am slightly worried about the logic between the two drifting apart. Ditto for `closed_curly_linter()`. Is there any tests we could set up to prevent that?

Reply:
I've left those alone to ensure only the specific lints are produced by the old implementations.
The only behavioural change is `open_curly_linter()` linting superfluous whitespace where `brace_linter()` does not.

To make this sure, I copied all tests from the merged linters over to `tests/testthat/test-brace_linter.R`

Enabling `brace_linter()` to only produce selected old lints seems like unnecessary complication of the new linter to me. WDYT?

Reply:
SGTM. it is technically a sort of breaking change (no longer able to toggle the linters individually), though I'm not sure anyone actually needed that. let's be sure it's adequately conveyed in the NEWS -- perhaps as part of the NEWS issue

Reply:
Yeah, let's consolidate all the breaking changes and migration instructions in a central place to keep them consistent.